code/ship/ship.cpp

```diff
@@ -5165,9 +5162,8 @@ void ship_init()
 			ships_inited = 1;
 		}
 
-		// NULL out "dynamic" subsystem ptr's
-		for (i = 0; i < NUM_SHIP_SUBSYSTEM_SETS; i++)
-			Ship_subsystems[i] = NULL;
+		// We shouldn't already have any subsystem pointers at this point.
+		Assertion(Ship_subsystems.empty(), "Some pre-allocated subsystems didn't get cleared out: %d batches present during ship_init(); get a coder!\n", Ship_subsystems.size());
```

Comment:
std::vector::size() returns a size_t but the format string expects an int. You can use the `SIZE_T_ARG` macro for cross-platform `size_t` formatting.